extensions/wikia/AchievementsII/services/AchAwardingService.class.php

```diff
@@ -42,10 +42,13 @@ public function migration($user_id)  {
 	 * @param Int $badge_type_id
 	 */
 	public function awardCustomNotInTrackBadge($user, $badge_type_id) {
-		wfProfileIn(__METHOD__);
+		global $wgExternalSharedDB, $wgAchievementsEditAddPhotoOnly;
 
-		global $wgExternalSharedDB;
+		if( !empty($wgAchievementsEditAddPhotoOnly) && ($badge_type_id != BADGE_EDIT && $badge_type_id != BADGE_PICTURE)) {
+			return;
+		}
 
+		wfProfileIn(__METHOD__);
```

Comment:
You shouldn't move wfProfileIn down - instead you should add another wfProfileOut